include/hip/hcc_detail/math_functions.h

```diff
@@ -20,237 +20,1125 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+__device__
+inline
+float atanf(float x) { return __ocml_atan_f32(x); }
+__device__
+inline
+float atanhf(float x) { return __ocml_atanh_f32(x); }
+__device__
+inline
+float cbrtf(float x) { return __ocml_cbrt_f32(x); }
+__device__
+inline
+float ceilf(float x) { return __ocml_ceil_f32(x); }
+__device__
+inline
+float copysignf(float x, float y) { return __ocml_copysign_f32(x, y); }
+__device__
+inline
+float cosf(float x) { return __ocml_cos_f32(x); }
+__device__
+inline
+float coshf(float x) { return __ocml_cosh_f32(x); }
+__device__
+inline
+float cospif(float x) { return __ocml_cospi_f32(x); }
+__device__
+inline
+float cyl_bessel_i0f(float x) { return __ocml_i0_f32(x); }
+__device__
+inline
+float cyl_bessel_i1f(float x) { return __ocml_i1_f32(x); }
+__device__
+inline
+float erfcf(float x) { return __ocml_erfc_f32(x); }
+__device__
+inline
+float erfcinvf(float x) { return __ocml_erfcinv_f32(x); }
+__device__
+inline
+float erfcxf(float x) { return __ocml_erfcx_f32(x); }
+__device__
+inline
+float erff(float x) { return __ocml_erf_f32(x); }
+__device__
+inline
+float erfinvf(float x) { return __ocml_erfinv_f32(x); }
+__device__
+inline
+float exp10f(float x) { return __ocml_exp10_f32(x); }
+__device__
+inline
+float exp2f(float x) { return __ocml_exp2_f32(x); }
+__device__
+inline
+float expf(float x) { return __ocml_exp_f32(x); }
+__device__
+inline
+float expm1f(float x) { return __ocml_expm1_f32(x); }
+__device__
+inline
+float fabsf(float x) { return __ocml_fabs_f32(x); }
+__device__
+inline
+float fdimf(float x, float y) { return __ocml_fdim_f32(x, y); }
+__device__
+inline
+float fdividef(float x, float y) { return x / y; }
+__device__
+inline
+float floorf(float x) { return __ocml_floor_f32(x); }
+__device__
+inline
+float fmaf(float x, float y, float z) { return __ocml_fma_f32(x, y, z); }
+__device__
+inline
+float fmaxf(float x, float y) { return __ocml_fmax_f32(x, y); }
+__device__
+inline
+float fminf(float x, float y) { return __ocml_fmin_f32(x, y); }
+__device__
+inline
+float fmodf(float x, float y) { return __ocml_fmod_f32(x, y); }
+__device__
+inline
+float frexpf(float x, int* nptr)
+{
+    int tmp;
+    float r =
+        __ocml_frexp_f32(x, (__attribute__((address_space(5))) int*) &tmp);
+    *nptr = tmp;
+
+    return r;
+}
+__device__
+inline
+float hypotf(float x, float y) { return __ocml_hypot_f32(x, y); }
+__device__
+inline
+int ilogbf(float x) { return __ocml_ilogb_f32(x); }
+__device__
+inline
+int isfinite(float x) { return __ocml_isfinite_f32(x); }
+__device__
+inline
+int isinf(float x) { return __ocml_isinf_f32(x); }
+__device__
+inline
+int isnan(float x) { return __ocml_isnan_f32(x); }
+__device__
+inline
+float j0f(float x) { return __ocml_j0_f32(x); }
+__device__
+inline
+float j1f(float x) { return __ocml_j1_f32(x); }
+__device__
+inline
+float jnf(int n, float x)
+{   // TODO: we could use Ahmes multiplication and the Miller & Brown algorithm
```

Comment:
We could do a lot of things with varying accuracy and performance.  Better just comment for now that this is just a placeholder until we see some demand and have some accuracy requirements.

Reply:
@b-sumner This pretty much applies to all other items - yes. However, these wonderful things have no place in the HIP RT, but rather should be in OCML. Ideally, at some point in the future, __ocml_jn_f32(...) will be available, and the HIP RT will just add a forward declare and call that.

Reply:
Agreed.